src/Context.tsx

```diff
@@ -0,0 +1,234 @@
+import * as Yup from "yup"
+import React, {
+  createContext,
+  useCallback,
+  useContext,
+  useEffect,
+  useState,
+} from "react"
+import uniqBy from "lodash.uniqby"
+
+export type ProgressiveOnboardingKey = ContextProps["keys"][number]
+export type ProgressiveOnboardingKeys = ProgressiveOnboardingKey[]
+
+interface DismissedKey {
+  key: ProgressiveOnboardingKey
+  timestamp: number
+}
+
+interface DismissedKeyStatus {
+  status: boolean
+  timestamp: number
+}
+
+export interface ContextProps {
+  dismissed: DismissedKey[]
+  dismiss: (
+    key: ProgressiveOnboardingKey | readonly ProgressiveOnboardingKey[]
+  ) => void
+  isDismissed: (key: ProgressiveOnboardingKey) => DismissedKeyStatus
+  keys: string[]
```

Comment:
IMO this should be a generic

Reply:
Totally, was planning on updating this before merging this PR 👍

Reply:
(gonna come back to this in a follow-up)